src/java/org/apache/cassandra/index/sai/plan/QueryController.java

```diff
@@ -765,10 +765,38 @@ else if (command instanceof MultiRangeReadCommand)
      *
      * @param iterator iterator over the keys from the index(es)
      */
-    float estimateSelectivity(RangeIterator iterator)
+    private float estimateSelectivity(RangeIterator iterator)
     {
-        float selectivity = Math.min((float) iterator.getMaxKeys() / estimateTotalAvailableRows(), 1.0f);
-        queryContext.setPostFilterSelectivityEstimate(selectivity);
+        long totalRowCount = estimateTotalAvailableRows();
```

Comment:
If `totalRowCount` is 0, which it can be for disjoint intersections, we will get an `NaN`, which triggers an exception. See this stack trace:

```
Uncaught exception on thread Thread[#749,ReadStage-1,5,main]
java.lang.IllegalArgumentException: perItemSuccessRate must not be NaN
at org.apache.cassandra.index.sai.utils.SoftLimitUtil.softLimit(SoftLimitUtil.java:49)
at org.apache.cassandra.index.sai.plan.QueryController.estimateSortThenFilterCost(QueryController.java:328)
at org.apache.cassandra.index.sai.plan.QueryController.decideFilterSortOrder(QueryController.java:303)
at org.apache.cassandra.index.sai.plan.QueryController.buildScoredPrimaryKeyIterator(QueryController.java:288)
at org.apache.cassandra.index.sai.plan.StorageAttachedIndexSearcher.buildScoredPrimaryKeyIterator(StorageAttachedIndexSearcher.java:147)
...
```

Reply:
I was slightly off. It's not a disjoint intersection, but an empty table. Here is a test that fails:

```java
    @Test
    public void testQueryOnEmptyTable() throws Throwable
    {
        createTable("CREATE TABLE %s (pk int, val text, vec vector<float, 2>, PRIMARY KEY(pk))");
        createIndex("CREATE CUSTOM INDEX ON %s(vec) USING 'StorageAttachedIndex'");
        createIndex("CREATE CUSTOM INDEX ON %s(val) USING 'StorageAttachedIndex'");
        waitForIndexQueryable();

        beforeAndAfterFlush(() -> {
            assertRows(execute("SELECT pk FROM %s WHERE val = 'A' ORDER BY vec ANN OF [1,1] LIMIT 1"));
        });
    }
```

Reply:
I'll take care of this in another PR.